### PR TITLE
ci: Post Integration Tests Trigger Comment on Base Branch Change

### PR DESCRIPTION
**Pull request by Muneeb147**
## Description
**Issue:** Integration tests were not triggered when base branch is changed from some-branch to master. Neither the PR author is aware that we have to re-trigger them.

**Solution:**
Wrote a github action which will subscribe "- edited" action and will be triggered on base branch change. As a result, a comment will be posted on PR notifying author to trigger integration tests.

**Why not do it programmatically? Other Solutions/Approaches**?
Such concerns are addressed in detail in following document:
https://docs.google.com/document/d/18cjQINkZ7Kc8eEGt0FrfuO78yShZAV32fnhfpG4A-9o/edit

### CODE REVIEW

File: .github/workflows/comment-on-base-branch-change.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+name: Post Integration Test Comment on Base Branch Change
+
+on:
+  pull_request:
+    types:
+      - edited
```
Comment (toliaqat) [inline .github/workflows/comment-on-base-branch-change.yml:6]: Will we get this message everytime we make any new commit or update anything in the PR while being based on master all the time?

Comment (Muneeb147) [reply .github/workflows/comment-on-base-branch-change.yml:6]: So edited event is triggered on following things:
- title change
- description change
- base branch change

We will only get this message when our base branch is changed from some X branch to "master". The following check in the action is making it sure.

```
if: ${{ github.event.changes.base && github.event.pull_request.base.ref == 'master' }}
```

File: .github/workflows/comment-on-base-branch-change.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    steps:
+      - name: Post a comment on the PR
+        uses: actions/github-script@v6
+        with:
+          script: |
+            github.rest.issues.createComment({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              issue_number: context.payload.pull_request.number,
+              body: "Base branch is changed to master. Please re-run the integration tests by adding "force:integration" label."
```
Comment (Muneeb147) [inline .github/workflows/comment-on-base-branch-change.yml]:
```suggestion
              body: "Base branch is changed to master. Please re-run the integration tests by adding 'force:integration' label."
```

File: .github/workflows/comment-on-base-branch-change.yml (hunk 1 above)
Comment (Muneeb147) [reply .github/workflows/comment-on-base-branch-change.yml:6]: By default github doesn't give any native event for base branch. We have to subscribe on "edited" and cover base change through if condition.

**Review by toliaqat — APPROVED** [review]

**Review by rabi-siddique — APPROVED** [review]

**Pull request merged**